src/main/java/cz/upce/fei/inptp/zz/entity/Vehicle.java

```diff
@@ -10,22 +10,29 @@
  * @author Roman
  */
 public class Vehicle {
-    private String identifier; // id should not be null
+    private String identifier;
     private int capacity; // capacity should be nonzero and positive number
     private VehicleType type;
     // TODO:initial position?
 
     public Vehicle(String identifier, int capacity, VehicleType type) {
-        this.identifier = identifier;
+        setIdentifier(identifier);
         this.capacity = capacity;
         this.type = type;
     }
-
-    
+  
     public String getIdentifier() {
         return identifier;
     }
 
+    private void setIdentifier(String identifier) {
+        if ((identifier == null) || (identifier.isEmpty())) {
```

Comment:
use java library - https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-
and null and empty are two different states